client/core/bookie.go

```diff
@@ -591,14 +594,46 @@ func handleTradeResumptionMsg(c *Core, dc *dexConnection, msg *msgjson.Message)
 	return nil
 }
 
-func (dc *dexConnection) refreshServerConfig() error {
+// refreshServerConfig fetches and replaces server configuration data. It also
+// initially checks that a server's API version is one of apiVers.
+func (dc *dexConnection) refreshServerConfig(apiVers []int) error {
 	// Fetch the updated DEX configuration.
 	cfg := new(msgjson.ConfigResult)
 	err := sendRequest(dc.WsConn, msgjson.ConfigRoute, nil, cfg, DefaultResponseTimeout)
 	if err != nil {
 		return fmt.Errorf("unable to fetch server config: %w", err)
 	}
 
+	// Check that we are able to communicate with this DEX.
+	apiVer := atomic.LoadInt32(&dc.apiVer)
+	cfgAPIVer := int32(cfg.APIVersion)
+
+	if apiVer != cfgAPIVer {
+		// If not initiation, do not allow api changes
+		// between connections.
+		if apiVer != -1 {
+			return fmt.Errorf("server API version unexpectedly changed from %v to %v",
```

Comment:
I've been thinking about this, and this seems to be inevitable, so I don't think we have a choice but to go along with the new version if it's recognized.  A server update that changes API version _will_ happen and I don't think we should always torpedo connected clients of they recognize the new api version.

Reply:
I agree. I just think it opens some angles possibly, that are hard to notice. Will remove, and we'll just be careful when updating.

Reply:
K. A warning would probably be good though.

Reply:
Added.